src/client/dfuse/pil4dfs/dfs_dcache.c

```diff
@@ -0,0 +1,663 @@
+/**
+ * (C) Copyright 2022-2024 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#define D_LOGFAC DD_FAC(il)
+
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <daos/debug.h>
+#include <gurt/hash.h>
+#include <gurt/list.h>
+#include <daos_fs.h>
+#include <daos/common.h>
+
+#include "dfs_dcache.h"
+
+/** Size of the hash key prefix */
+#define DCACHE_KEY_PREF_SIZE 35
+#define DCACHE_KEY_MAX       (DCACHE_KEY_PREF_SIZE - 1 + PATH_MAX)
+
+#ifdef DAOS_BUILD_RELEASE
+
+#define DF_DK          "dk[%zi]"
+#define DP_DK(_dk)     strnlen((_dk), DCACHE_KEY_MAX)
+
+#define DF_PATH        "path[%zi]"
+#define DP_PATH(_path) strnlen((_path), PATH_MAX)
+
+#else
+
+#define DF_DK          "dk'%s'"
+#define DP_DK(_dk)     (_dk)
```

Comment:
It's worth noting that the `DF_DE` macro will mask out ` from the logs so they can be parsed by tools correctly.

Reply:
Just a problem of size in release mode with strnlen